kit/Kit.cpp

```diff
@@ -894,7 +876,6 @@ class Document final : public DocumentManagerInterface
 
         std::shared_ptr<ChildSession> session = it->second;
 
-        // Flag and release lock.
         ++_isLoading;
```

Comment:
I imagine `_isLoading` would need to go too. If that's too large a change in this PR, then we can leave it for a later patch (and revert the relevant changes in this function/file).

Reply:
okay, so I will work with isloading on another patch.

Reply:
@22shubh22 good, but you have already made changes that affect `_isLoading`. Best to revert those if you will remove _isLoading in another patch. I fear the partial change might cause some issues.

Shouldn't be difficult to find the related code that I'm referring to, but feel free to ask if you need help.

Thanks again!